packages/shell-api/src/decorators.ts

```diff
@@ -362,11 +406,20 @@ export function topologies(topologiesArray: Topologies[]): Function {
 }
 export const nonAsyncFunctionsReturningPromises: string[] = []; // For testing.
 export function returnsPromise(_target: any, _propertyKey: string, descriptor: PropertyDescriptor): void {
-  const orig = descriptor.value;
-  orig.returnsPromise = true;
-  descriptor.value = markImplicitlyAwaited(descriptor.value);
-  if (orig.constructor.name !== 'AsyncFunction') {
-    nonAsyncFunctionsReturningPromises.push(orig.name);
+  const originalFunction = descriptor.value;
+  originalFunction.returnsPromise = true;
+
+  async function wrapper(this: any, ...args: any[]) {
+    const result = await originalFunction.call(this, ...args);
+    await new Promise(setImmediate);
+    return result;
```

Comment:
Maybe this, so that a) we always await the tick, even if it throws, and b) the java-shell is not affected by this?

```suggestion
    try {
      return await originalFunction.call(this, ...args);
    } finally {
       if (typeof setImmediate === 'function') {
         // Not all JS environments have setImmediate
         await new Promise(setImmediate);
       }
    }
```

(I’ll apply the suggestion, but I’ll leave the comment so you can see what I’ve changed here :))